### PR TITLE
RFC: Use Vec<char> for input instead of String

### PR DESCRIPTION
**Pull request by fangyi-zhou**
This may make it easier to handle inputs since we do not have to care
about character boundaries in Strings.

See #30

I'm not sure whether this is a good idea performance-wise, I would like
to hear your opinions

### CODE REVIEW

File: src/handlers/input.rs [hunk 1]
```diff
@@ -26,15 +27,15 @@ pub fn handler(key: Key, app: &mut App) {
         }
         Key::Left => {
             if !app.input.is_empty() && app.input_idx > 0 {
-                let last_c = app.input.chars().nth(app.input_idx - 1).unwrap();
+                let last_c = app.input[app.input_idx - 1];
```
Comment (Rigellute) [inline src/handlers/input.rs]: Could there ever be a situation where `app.input_idx - 1` goes out of bounds?

Perhaps it's safer to do `app.input.get(app.input_idx - 1)` and then handle the `Option`?

File: src/handlers/input.rs [hunk 2]
```diff
@@ -26,15 +27,15 @@ pub fn handler(key: Key, app: &mut App) {
...
                 app.input_idx -= 1;
                 let width: u16 = UnicodeWidthChar::width(last_c).unwrap().try_into().unwrap();
                 app.input_cursor_position -= width;
             }
         }
         Key::Right => {
             if app.input_cursor_position < app.input.len().try_into().unwrap() {
-                let next_c = app.input.chars().nth(app.input_idx).unwrap();
+                let next_c = app.input[app.input_idx];
```
Comment (Rigellute) [inline src/handlers/input.rs]: Same as above

File: src/handlers/input.rs (hunk 1 above)
Comment (fangyi-zhou) [reply src/handlers/input.rs]: it is guarded by the condition `app.index_idx > 0` so it's safe

File: src/handlers/input.rs [hunk 3]
```diff
@@ -26,15 +27,15 @@ pub fn handler(key: Key, app: &mut App) {
...
             if !app.input.is_empty() && app.input_idx > 0 {
-                let last_c = app.input.chars().nth(app.input_idx - 1).unwrap();
+                let last_c = app.input[app.input_idx - 1];
                 app.input_idx -= 1;
                 let width: u16 = UnicodeWidthChar::width(last_c).unwrap().try_into().unwrap();
                 app.input_cursor_position -= width;
             }
         }
         Key::Right => {
             if app.input_cursor_position < app.input.len().try_into().unwrap() {
```
Comment (fangyi-zhou) [inline src/handlers/input.rs]:
```suggestion
            if app.input_idx < app.input.len() {
```

**Comment by Rigellute** [comment]
@all-contributors please add @fangyi-zhou for code

**Pull request merged**